update/delta/delta_1.3.7_001_search_opened_valves.sql

```diff
@@ -0,0 +1,207 @@
+alter table qwat_network.network add column if not exists network_id serial ;
+
+/*
+Check if a node is a valve
+Is the parameter _check_if_network_function is true, then we also check if the valve is a network valve
+*/
+CREATE OR REPLACE FUNCTION qwat_network.ft_check_node_is_valve(_id integer, _check_if_network_function boolean)
+ RETURNS boolean
+ LANGUAGE plpgsql
+AS $function$
+declare 
+    count integer := 0;
+    is_network boolean := false;
+begin
+    
+    --raise notice 'vanne %', _id;
+--     if _id = 54387 then
+--         raise notice 'vanne 54387 %', _check_if_network_function;
+--     end if;
+
+    select count(*) into count from qwat_od.valve where id = _id;
+    if count > 0 then
+        if _check_if_network_function then
+            is_network := (select qwat_network.ft_check_valve_is_network(_id));
+            if is_network then
+                --raise notice '---------- vanne réseau';
+                return true;
+            else
+                return false;
+            end if;
+        end if;
+        return true;
+    end if;
+    return false;
+end
+$function$
+;
+
+/*
+Check if a valve is a network valve
+*/
+CREATE OR REPLACE FUNCTION qwat_network.ft_check_valve_is_network(_id integer)
+ RETURNS boolean
+ LANGUAGE plpgsql
+AS $function$
+declare 
+    _function integer;
+begin
+    select fk_valve_function into _function from qwat_od.valve where id = _id;
+    if _function != 6105 and _function != 6108 then
+        return true;
+    end if;
+    return false;
+end
+$function$
+;
+
+
+CREATE OR REPLACE FUNCTION qwat_network.ft_search_opened_valves(start_pipe integer, _x float, _y float, max_km integer DEFAULT 20, stop_on_network_valves boolean DEFAULT true)
+ RETURNS TABLE(id integer, geometry geometry)
+ LANGUAGE plpgsql
+AS $function$
+declare 
+    valve integer;
+    pipe integer;
+    start_node integer;
+    start_node2 integer;
+    network_id integer;
+    valve_list integer[]='{0}';
+    rec record;
+    res boolean;
+    sql text;
+    crs integer;
+begin
+	-- Get CRS from qwat_sys.settings
+	sql = format('select value from qwat_sys.settings;');
+	execute sql into crs;
+    -- Select a network object from the pipe id and point clicked
+    sql = format('
+        with point_clicked as ( 
+            select st_setsrid(st_makepoint($1,$2),$4) as geom
+        )
+        select n.network_id, n.source, n.target
+        from qwat_network.network n, point_clicked 
+        where n.id = $3
+        and st_dwithin(point_clicked.geom, n.geometry, 1);'); -- 1m is enough ?
```

Comment:
the tolerance should be a parameter of the function, no? so it can be adjusted from the client depending on the scale of the map.